### ISSUE TITLE
Could not find a declaration file for module 'filedirname'

### ISSUE THREAD

**Issue by gfortaine** [bevry/filedirname#224]
This bug appears with the following options in `tsconfig` ([TS >= 4.7](https://devblogs.microsoft.com/typescript/announcing-typescript-4-7-beta/#package-json-exports-imports-and-self-referencing)) :

```json
{
  "compilerOptions": {
    "strict": true,
    "moduleResolution": "nodenext",
    "allowJs": false,
    ...
  }
}
```

Here is a sample repo to reproduce : https://github.com/gfortaine/filedirname/tree/fix/moduleResolution-nodenext

```shell
$ npm install
$ npm run build:bug
```

**Comment by balupton** [comment]
I tried to reproduce this locally but can't.

**Comment by balupton** [comment]
Was able to reproduce, details over at
https://github.com/bevry/boundation/issues/158#issuecomment-1819097797

**Comment by balupton** [comment]
this should now be resolved with v2.11.0
https://github.com/bevry/filedirname/commit/48958c762bc1a70683de1e40026114745ac960bb#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519

shoutout to my [GitHub Sponsors](https://github.com/sponsors/balupton) for affording me the time to make this happen
